Design/Usage/AlgorithmUsageStatistics.md

```diff
@@ -0,0 +1,148 @@
+
+Initially we do not plan to define reports for the feature usage data, they will be queried direct from the database using SQL.
+Once a good understanding of the usefull aspects of the data clarifies we will look to integrating some reports into the API.
+
+#####Database size considerations
+
+The current size of the gears instance we are using is roughly 300MB, of which the Sevices_Usage table (the startup usage table) is 30MB, when populated with roughly 1.5 years of data.  Given that we can expect 10-100 times as much data in the Features usage table then this will make the table 300MB to 3GB in size within 1.5 years.
+
+The current free plan we use allows for 1GB of storage, however the Bronze plan allows for 1GB of free storage, followed by $1 per month for each additional GB (up to 30GB max).  Once we get near to the end of the free plan we would move to the bronze plan and pay the small fee.
+
+The value of old feature usage data over 1 year rapidly diminshes, so we would plan to remove old data on a yearly basis.
+
+###To Existing Code
+
+####API::FrameworkManager
+
+Currently the code to send usage reports is in the FrameworkManager (FrameworkManagerImpl::SendStartupUsageInfo).  This would be removed.
+
+
+####Kernel::ConfigService
+
+As described above this change would create a UsageReporter class which would be responsible for collating, and sending
+all usage data.   This will be created, and owened by UsageReporter.registerStartup().
+The config service will also delete the UsageReporter as the first step of its desctutor, when logging and all of the other ConfigService methods remain available.
+
+###API::Algorithm
+
+We would add a call to UsageReporter.registerFeatureUsage() within the Execute method of the Algorithm base class,
+after exec has been called, close to where we record history.
+This would simply record the information to be sent into a local queue object and return quickly.
+
+
+###Algorithm::SendUsage
+
+In order to harmonise the code and keep things simple and easy to follow, this algorithm will be removed and the functionality added to the UsageReporter.  As part of this the current hand crafted json creation will be repleaced using jsoncpp.
+
+#Apendix
+##Json for startup calls
+``` json
+{  
+   "ParaView":0,
+   "application":"mantidplot",
+   "dateTime":"2015-11-30T16:20:39.060539000",
+   "host":"df2545f221f5cecc6752219e1716384a",
+   "mantidSha1":"a2c602fb1f8cb339abed583bc1e3e6af992ea9db",
+   "mantidVersion":"3.5.20151127.836",
+   "osArch":"AMD64",
+   "osName":"Windows NT",
+   "osReadable":"Microsoft Windows 7 Professional",
+   "osVersion":"6.1 (Build 7601: Service Pack 1)",
+   "uid":"114ede53ec4e133a0d637f889ef8764d"
+} 
+```
+##Json for feature usage calls
+Note: This has an almost identical header than that of the startup calls.
+``` json
+{  
+   "ParaView":0,
+   "dateTime":"2015-11-30T16:22:35.270917000",
+   "features":[  
```

Comment:
I'm not comfortable with centrally logging all of the parameter values. It also gets "big" when the parameter is a `FitFunction` and useless when the parameter is a `PropertyManager`.